### PR TITLE
fix returned ledger_index_min and ledger_index_max bug in account_tx

### LINKED ISSUE #217: account_tx clio response ledger_index_max and ledger_index_min values differ from rippled response.

### PR DESCRIPTION
**Pull request by cjcobb23**
Fixes https://github.com/XRPLF/clio/issues/217

- - -
This change is [Reviewable](https://reviewable.io/reviews/xrplf/clio/258)

### CODE REVIEW

**Review by natenichols — CHANGES_REQUESTED** [review]
I think this is wayyyyy easier than we're making it:

I ran 
```
{
  "id": 2,
  "command": "account_tx",
  "account": "rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn",
  "ledger_index_min": -1,
  "ledger_index_max": -1,
  "binary": false,
  "limit": 2,
  "forward": false
}
```

And got 
```
    "ledger_index_max": 73893759,
    "ledger_index_min": 32570,
    "limit": 2,
    "marker": {
      "ledger": 69841511,
      "seq": 82
    },
```

So I think the response literally just returns `minIndex` and `maxIndex` even if there are more to search.

**Comment by cjcobb23** [comment]
It's supposed to return the ledger range actually searched, per the documentation

**Comment by cjcobb23** [comment]
```
ledger_index_min | Integer - Ledger Index | The ledger index of the earliest ledger actually searched for transactions.
ledger_index_max | Integer - Ledger Index | The ledger index of the most recent ledger actually searched for transactions.

```

https://xrpl.org/account_tx.html#response-format

**Comment by natenichols** [comment]
Yea, but rippled doesn't look like it actually does that...

**Comment by natenichols** [comment]
So do we want to match behavior in the docs or behavior in rippled

**Comment by cjcobb23** [comment]
Docs, since thats what devs refer to.

**Comment by cjcobb23** [comment]
Trying to match rippled is a whole can of worms. What if rippled has a bug? Do we match the bug? What if reporting mode and rippled differ slightly? What if some things can't reasonably be mapped 1:1, like markers?

**Review by natenichols — APPROVED** [review]

**Comment by natenichols** [comment]
You may also wanna ping Greg so he is aware of these changes as he works on the `nft_tx` PR.

File: src/rpc/handlers/AccountTx.cpp [hunk 1]
```diff
@@ -188,16 +188,43 @@ doAccountTx(Context const& context)
...
     {
-        response[JS(ledger_index_max)] = cursor->ledgerSequence;
-        response[JS(ledger_index_min)] = minIndex;
+        // min of returned cursor and max index
+        response[JS(ledger_index_max)] = retCursor
+            ? std::min(retCursor->ledgerSequence, maxIndex)
+            : maxIndex;
+        // max of input cursor and min index
+        response[JS(ledger_index_min)] =
+            cursor ? std::max(cursor->ledgerSequence, minIndex) : minIndex;
```
Comment (ledhed2222) [inline src/rpc/handlers/AccountTx.cpp:227]: i don't believe the API on rippled is this complex actually. I _believe_ it just returns the parsed values from the input `legder_index_min/max`. In other words, you don't have to take into account the marker/cursor, just the min/max indexes of the ledger.

i think it's as simple as:

```c++
response[JS(ledger_index_max)] = maxIndex;
response[JS(ledger_index_min)] = minIndex;
```

for all situations, including with a marker and for reverse search.

Comment (ledhed2222) [reply src/rpc/handlers/AccountTx.cpp:227]: here's some evidence, from the account_tx gui on xrpl.org:

request:

```
{
  "id": 2,
  "command": "account_tx",
  "account": "rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn",
  "ledger_index_min": -1,
  "ledger_index_max": -1,
  "binary": true,
  "limit": 2,
  "forward": false,
  "marker": {
    "ledger": 69841511,
    "seq": 82
  }
}
```

response:

```
{
  "id": 2,
  "result": {
    "account": "rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn",
    "ledger_index_max": 73897122,
    "ledger_index_min": 32570,
    "limit": 2,
    "marker": {
      "ledger": 68558626,
      "seq": 37
    },
    "transactions": [
      {
        "ledger_index": 69841511,
        "meta": "201C00000052601240000000004C4B40F8E511006125041CD7E1558566673ECD0A9731C516906E5D2F47129C5C13713602140733831A56CEAE1A055613F1A95D7AAB7108D5CE7EEAF504B2894B8C674E6D68499076441C4837282BF8E662400000003CE7B06DE1E7220092000024000001812BFFFFFFFF2D0000000C4198B4375E1D753E5B91627516F6D70977594E0AA11CBDD1760DE95B68DF2ABBE75C9698CEB548BEA9789053FCB3EBD444FB62400000003D33FBAD72110000000000000000000000070000000300770A6D64756F31332E636F6D81144B4E9C06F24296074F7BC48F92A97916C6DC5EA98814854DFEFC8AA8C17FF2B705FEC5D55F933C9CCF1DE1E1E411006156377D6778741A746817C993E97BB57F35E528B582B77BE28F9397CBD84E6E5CB0E62403FEB6FD624000000000989680E1E722000000002403FEB6FE250429B2182D0000000055474E7682B4D9F2B752C7897C5C0362C93397C361D14A5A68F2A4BA2F06E54D7F6240000000000000008114FF7865649238DEB193152E3154A94EDC5E3C1F32E1E1F1031000",
        "tx_blob": "12001522800000002403FEB6FD2E0001E0F3201B0429B27A6840000000004C4B4073210263F02703A97E5BD3AA671183D83219522704CCC49D85EB1AC199989181D405A674473045022100A3AF36CB35476F22B83C2ECB7003D9C0BED72B201C102E774FB75E37908C9911022055C29E1C2CFBFF9610341B352E140AD59F5146BAD02AD90E44BDA05B877494768114FF7865649238DEB193152E3154A94EDC5E3C1F3283144B4E9C06F24296074F7BC48F92A97916C6DC5EA9",
        "validated": true
      },
      {
        "ledger_index": 68999137,
        "meta": "201C00000045F8E51100612504161F2255AC7EAAE8B0CB22FA837C5EFD74201E8294D1FF3925471093C53768395F1226925613F1A95D7AAB7108D5CE7EEAF504B2894B8C674E6D68499076441C4837282BF8E1E511006125041CD400559972671EF0FA6F2A6F9A0AF63E4CB6F1ABB345272856CB0745657A858E11F45B5624FC0F31D3D60D19431D27044E0B2F72DC57B611476D607B9D262F4CAFC2FEDEE624040DCBD02D0000000062400000000180DFCBE1E7220000000024040DCBD12D0000000162400000000180DFB7811460D80554EFC98021A7A5F5434631E55C17E33E5EE1E1E5110064563B9C0CE77FCE7BCEE1A68F1E26AC467AF326239D0D816CE705E4A0E2DAD03F6DE72200000000583B9C0CE77FCE7BCEE1A68F1E26AC467AF326239D0D816CE705E4A0E2DAD03F6D82144B4E9C06F24296074F7BC48F92A97916C6DC5EA9E1E1E31100645671CA0E9216D802A008583FD945623A4861EF6414B7C7C5447DD4D9309198A7D0E85871CA0E9216D802A008583FD945623A4861EF6414B7C7C5447DD4D9309198A7D0821460D80554EFC98021A7A5F5434631E55C17E33E5EE1E1E311007256965488EF00AE88F938930FC88774B13F99FFFA00883004AD18AF4F0EE1C7B26FE822000200006280000000000000000000000000000000000000005553440000000000000000000000000000000000000000000000000166800000000000000000000000000000000000000055534400000000004B4E9C06F24296074F7BC48F92A97916C6DC5EA967D5838D7EA4C68000000000000000000000000000555344000000000060D80554EFC98021A7A5F5434631E55C17E33E5EE1E1F1031000",
        "tx_blob": "120014220004000024040DCBD063D5838D7EA4C6800000000000000000000000000055534400000000004B4E9C06F24296074F7BC48F92A97916C6DC5EA96840000000000000147321029371A582A3F9D54B128B41E37CD325370512E0FB5215BD52A00E011B53749E2B74463044022006AB30FE425FD68528A5FFA7D1E027C14EEF2781106D9D03FA1E6839CD4AD39202200ECFDFDB86942437887D35D98AF9472C2FB40EA64886FF67724FB66A02EC29D8811460D80554EFC98021A7A5F5434631E55C17E33E5E",
        "validated": true
      }
    ],
    "used_postgres": true,
    "validated": true,
    "warnings": [
      {
        "id": 1004,
        "message": "This is a reporting server.  The default behavior of a reporting server is to only return validated data. If you are looking for not yet validated data, include \"ledger_index : current\" in your request, which will cause this server to forward the request to a p2p node. If the forward is successful the response will include \"forwarded\" : \"true\""
      }
    ]
  },
  "status": "success",
  "type": "response"
}
```

Comment (cjcobb23) [reply src/rpc/handlers/AccountTx.cpp:227]: rippled might have a bug, but i don't think that matters. We just have to match the docs, as that's what devs actually use to develop.

Comment (cjcobb23) [reply src/rpc/handlers/AccountTx.cpp:227]: the other option is to change the docs.

**Comment by cjcobb23** [comment]
I decided this logic is too complicated, and we should just mirror rippled

**Pull request closed without merging**